packages/webapp/src/components/Modals/QuickAssignModal/index.jsx

```diff
@@ -60,24 +113,134 @@ export default function TaskQuickAssignModal({
+          assignee_user_id: assigneeUserId,
         });
+
+    if (isYesOptionSelected(selectedHourlyWageAction)) {
+      const wage = +parseFloat(hourlyWage).toFixed(2);
+
+      if (selectedHourlyWageAction === hourlyWageActions.SET_HOURLY_WAGE) {
+        onUpdateUserFarmWage({ user_id: assigneeUserId, wage: { type: 'hourly', amount: wage } });
+      } else if (selectedHourlyWageAction === hourlyWageActions.FOR_THIS_TASK) {
+        onChangeTaskWage(wage);
+      }
+    } else if (selectedHourlyWageAction === hourlyWageActions.DO_NOT_ASK_AGAIN) {
+      onSetUserFarmWageDoNotAskAgain({ user_id: assigneeUserId });
+    }
+
     dismissModal();
   };
 
-  const onCheckedAll = () => {
-    setAssignAll(!assignAll);
-  };
+  const disabled = !selectedWorker || !isValid;
 
-  const disabled = selectedWorker === null;
+  useEffect(() => {
+    let shouldShowWageSection = false;
+
+    if (user.is_admin && selectedWorker) {
+      const assigned = selectedWorker.label !== unAssignedOption.label;
+
+      if (assigned) {
+        const { amount } = selectedWorker.wage;
+        const hasWage = !!(amount || wageAtMoment);
+        shouldShowWageSection = !hasWage && !selectedWorker.doNotAskAgain;
+      }
+    }
+    setShowHourlyWageSection(shouldShowWageSection);
+  }, [user.is_admin, selectedWorker, wageAtMoment]);
+
+  useEffect(() => {
+    if (!user.is_admin) {
+      return;
+    }
+
+    resetField(HOURLY_WAGE_ACTION);
+    resetField(HOURLY_WAGE);
+  }, [user.is_admin, selectedWorker]);
+
+  useEffect(() => {
+    let shouldShowWageInput = false;
+
+    if (selectedHourlyWageAction) {
+      shouldShowWageInput = isYesOptionSelected(selectedHourlyWageAction);
+    }
+    if (!shouldShowWageInput) {
+      resetField(HOURLY_WAGE);
+    }
+
+    setShowHourlyWageInput(shouldShowWageInput);
+  }, [selectedHourlyWageAction]);
+
+  const radioOptions = [
+    {
+      label: t('ADD_TASK.HOURLY_WAGE.SET_HOURLY_WAGE'),
+      value: hourlyWageActions.SET_HOURLY_WAGE,
+    },
+    {
+      label: t('ADD_TASK.HOURLY_WAGE.FOR_THIS_TASK'),
+      value: hourlyWageActions.FOR_THIS_TASK,
+    },
+    {
+      label: t('common:NO'),
+      value: hourlyWageActions.NO,
+    },
+    {
+      label: t('ADD_TASK.HOURLY_WAGE.DONT_ASK'),
+      value: hourlyWageActions.DO_NOT_ASK_AGAIN,
+    },
+  ];
+
+  const renderHourlyWageSection = () => {
+    if (!showHourlyWageSection) {
+      return;
+    }
+
+    return (
+      <>
+        <Label className={styles.warning} style={{ marginBottom: 24 }}>
+          {t('ADD_TASK.HOURLY_WAGE.ASSIGNEE_WAGE_WARNING', { name: selectedWorker.label })}
+        </Label>
+        <Main style={{ marginBottom: 10 }}>
+          {t('ADD_TASK.HOURLY_WAGE.WANT_TO_SET_HOURLY_WAGE')}
+        </Main>
+        <RadioGroup
+          hookFormControl={control}
+          name={HOURLY_WAGE_ACTION}
+          radios={radioOptions}
+          data-cy="quickAssign-hourlyWageAction"
+          style={{ marginBottom: 10 }}
```

Comment:
I am personally open to changing this standard but most of the app explicitly states the units 'px'

Reply:
I prefer to take advantage of not having to type `''` and units, but I'll follow what's been done for now.